lib/audited/auditor.rb

```diff
@@ -232,7 +243,23 @@ def require_comment
       end
 
       def auditing_enabled
-        self.class.auditing_enabled
+        if_condition = audited_options[:if]
+        unless_condition = audited_options[:unless]
+        return self.class.auditing_enabled unless if_condition.present? || unless_condition.present?
+
+        return run_conditional_check(audited_options[:if]) &&
+          run_conditional_check(audited_options[:unless], matching: false) &&
+          self.class.auditing_enabled
+      end
+
+      def run_conditional_check(condition, matching: true)
+        return true if condition.respond_to?(:blank?) && condition.blank?
```

Comment:
`blank?` is defined by activesupport on `Object`, so no need to check if the condition responds to it first.